core/src/main/java/org/mule/runtime/core/internal/el/DefaultBindingContextBuilder.java

```diff
@@ -125,14 +129,14 @@ public DefaultBindingContextBuilder(BindingContext bindingContext) {
 
   @Override
   public BindingContext build() {
-    return new BindingContextImplementation(parent, unmodifiableMap(bindings),
+    return new BindingContextImplementation(new ArrayList<>(delegates), unmodifiableMap(bindings),
```

Comment:
Why the new array list?

Reply:
because the list in the builder would still be mutable, and being a `LinkedList`, it is cheaper to build an `ArrayList` than to use a `Collections.unmodifiableList`